pkg/operator/starter.go

```diff
@@ -160,7 +154,10 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 	if err != nil {
 		return err
 	}
-	migrator := migrators.NewInProcessMigrator(dynamicClientForMigration, kubeClient.Discovery())
+
+	migrationClient := kubemigratorclient.NewForConfigOrDie(controllerContext.KubeConfig)
+	migrationInformer := migrationv1alpha1informer.NewSharedInformerFactory(migrationClient, time.Minute*30)
+	migrator := migrators.NewKubeStorageVersionMigrator(migrationClient, migrationInformer.Migration().V1alpha1(), kubeClient.Discovery())
```

Comment:
@sttts we should switch to `dynamic-{informer,lister,client}` so that we don't depend on `kubernetes-sigs/kube-storage-version-migrator` in `cluster-kube-apiserver-operator`

Reply:
Ugly. I see some risk that we might depend on upstream's kube version level though. Ugly, did I mention that before?

Reply:
@vareti can you prototype that?

Reply:
@p0lyn0mial not a blocker, is it?

Reply:
no, not a blocker.

Reply:
@sttts yes, I will try to prototype this tomorrow.